src/formatDate.js

```diff
@@ -50,7 +50,46 @@
  */
 
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const dateArr = date.split(fromFormat[3]);
+  const DAY_IDENTIFIER = 'DD';
+  const MONTH_IDENTIFIER = 'MM';
+  const YEAR_SHORT_IDENTIFIER = 'YY';
+  const YEAR_LONG_IDENTIFIER = 'YYYY';
+  let result = toFormat.slice(0, 3);
+
+  result[result.indexOf(DAY_IDENTIFIER)]
+  = dateArr[fromFormat.indexOf(DAY_IDENTIFIER)];
+
+  result[result.indexOf(MONTH_IDENTIFIER)]
+  = dateArr[fromFormat.indexOf(MONTH_IDENTIFIER)];
+
+  if (fromFormat.includes(YEAR_SHORT_IDENTIFIER)
+      && result.includes(YEAR_SHORT_IDENTIFIER)) {
+    result[result.indexOf(YEAR_SHORT_IDENTIFIER)]
+    = dateArr[fromFormat.indexOf(YEAR_SHORT_IDENTIFIER)];
+  } else if (fromFormat.includes(YEAR_LONG_IDENTIFIER)
+              && result.includes(YEAR_LONG_IDENTIFIER)) {
+    result[result.indexOf(YEAR_LONG_IDENTIFIER)]
+          = dateArr[fromFormat.indexOf(YEAR_LONG_IDENTIFIER)];
+  } else if (fromFormat.includes(YEAR_SHORT_IDENTIFIER)
+              && result.includes(YEAR_LONG_IDENTIFIER)) {
+    if (dateArr[fromFormat.indexOf(YEAR_SHORT_IDENTIFIER)] >= 30) {
+      result[result.indexOf(YEAR_LONG_IDENTIFIER)]
+            = '19' + dateArr[fromFormat.indexOf(YEAR_SHORT_IDENTIFIER)];
+    } else {
+      result[result.indexOf(YEAR_LONG_IDENTIFIER)]
+            = '20' + dateArr[fromFormat.indexOf(YEAR_SHORT_IDENTIFIER)];
+    }
+  } else if (fromFormat.includes(YEAR_LONG_IDENTIFIER)
+              && result.includes(YEAR_SHORT_IDENTIFIER)) {
+    result[result.indexOf(YEAR_SHORT_IDENTIFIER)]
+  = (dateArr[fromFormat.indexOf('YYYY')][2]
+  + dateArr[fromFormat.indexOf('YYYY')][3]);
+  }
```

Comment:
Instead of using multiple conditional blocks, consider a more streamlined approach. Use loops or simpler conditional statements to handle the transformation.

You can create an object that correlates the elements in the fromFormat and toFormat arrays. This object will help in mapping the elements directly without numerous conditional checks. For example 

```
 const keysDates = {};

 for (let i = 0; i < fromFormat.length - 1; i++) {
    keysDates[fromFormat[i]] = dateParts[i];
  }
```